pyinterprod/uniprot/unirule.py

```diff
@@ -707,3 +707,221 @@ def update_signatures(filepath: str, uri: str):
+        best_subgroup = None
+        for subgroup in subgroups:
+            coverage = set()
+            pfams = 0
+            _subgroup = []
+
+            for i in subgroup:
+                domain = group[i]
+                coverage |= domain["residues"]
+                if domain["rank"] == 0:
+                    pfams += 1
+
+                _subgroup.append(domain)
+
+            coverage = len(coverage)
+            if coverage < max_coverage:
+                continue
+            elif coverage > max_coverage or pfams > max_pfams:
+                max_coverage = coverage
+                max_pfams = pfams
+                best_subgroup = _subgroup
+
+        # Flag selected representative domains
+        for domain in best_subgroup:
+            domain["representative"] = True
+    return domains
+
+
+def _resolve_domains(graph: dict[int, set[int]]) -> list[set[int]]:
+    def is_valid(candidate: list[int]) -> bool:
+        for node_a in candidate:
+            for node_b in candidate:
+                if node_a != node_b and node_a not in graph[node_b]:
+                    return False
+
+        return True
+
+    def make_sets(current_set: list[int], remaining_nodes: list[int]):
+        if is_valid(current_set):
+            if not remaining_nodes:
+                all_sets.append(set(current_set))
+                return True
+        else:
+            return False
+
+        current_node = remaining_nodes[0]
+        remaining_nodes = remaining_nodes[1:]
+
+        # Explore two possibilities at each step of the recursion
+        # 1) current node is added to the set under consideration
+        make_sets(current_set + [current_node], remaining_nodes)
+        # 2) current node is not added to the set
+        make_sets(current_set, remaining_nodes)
+
+    all_sets = []
+    make_sets([], list(graph.keys()))
+    return all_sets
+
+
+def _calc_coverage(domain: dict) -> set[int]:
+    residues = set()
+    for f in domain["fragments"]:
+        residues |= set(range(f["start"], f["end"] + 1))
+
+    return residues
+
+
+def _eval_overlap(dom_a: dict, dom_b: dict, threshold: float) -> bool:
+    overlap = dom_a["residues"] & dom_b["residues"]
+    if overlap:
+        len_a = len(dom_a["residues"])
+        len_b = len(dom_b["residues"])
+        return len(overlap) / min(len_a, len_b) >= threshold
+
+    return False
+
+
+def _get_fragments(pos_start: int, pos_end: int, fragments: str) -> list[dict]:
+    if fragments:
+        result = []
+        for frag in fragments.split(','):
+            # Format: START-END-STATUS
+            s, e, t = frag.split('-')
+            result.append({
+                "start": int(s),
+                "end": int(e),
+                "dc-status": DC_STATUSES[t]
+            })
+
+        result.sort(key=lambda x: (x["start"], x["end"]))
+    else:
+        result = [{
+            "start": pos_start,
+            "end": pos_end,
+            "dc-status": DC_STATUSES['S']  # Continuous
```

Comment:
The character cote of is enough for `dc-status`:

```suggestion
                "dc-status": t
            })

        result.sort(key=lambda x: (x["start"], x["end"]))
    else:
        result = [{
            "start": pos_start,
            "end": pos_end,
            "dc-status": "S"  # Continuous
```